r/tests/testthat/test-matrix_stats.R

```diff
@@ -187,4 +187,62 @@ test_that("svds registers generic with RSpectra", {
   # Ensure RSpectra works with BPCells and BPCells works with RSpectra
   equal_svds(ans, RSpectra::svds(i1, k=5))
   equal_svds(ans, BPCells::svds(m1, k=5))
-})
\ No newline at end of file
+})
+
+test_that("row_max works for iterable matrices", {
+  withr::local_seed(195123)
+  m1 <- matrix(1, nrow = 3, ncol = 4)
+  i1 <- m1 %>% as("dgCMatrix") %>% as("IterableMatrix")
+  expect_equal(row_max(i1), matrixStats::rowMaxs(as.matrix(m1)))
+})
+
```

Comment:
Regarding the test cases:

1. The matrices are mostly initialized to constants, so using one of the helpers like `generate_sparse_matrix` or `generate_dense_matrix` might be useful to create slightly more complicated test cases.
2. One set of test areas that aren't addressed here is matrix rows/cols that are all negative, or are all negative/0. 

Finally, it might be nice to consolidate these tests a bit (though this is also a matter of personal preference). A recipe I like for these types of things is:

1. Make a hand-crafted test matrix that hits all the corner cases I'm worried about across its various rows.
2. Make a randomly-generated sparse matrix that's about 4 x 64, which should cover all sorts of sparsity patterns given that `generate_sparse_matrix` defaults to 50% sparsity.

Then for each of the matrices test that rowMaxs and colMaxs both work for the transpose and the original matrix (both transposing before and after converting to IterableMatrix). 
Putting those tests in a loop is an option if you have enough variants to test, though in this case it might not be needed.

My attitude on these tests is that the goal is to flag the presence of a problem but it's okay if it takes a bit more investigation to nail down the precise nature of the bug if it exists.